### ISSUE TITLE
Inconsistent use of `description` and `definition` in codes

### ISSUE THREAD

**Issue by luciecastella** [IAMconsortium/nomenclature#180]
Some projects use `description` and others `definition` as code attribute. This should maybe be unified. Which one is more relevant? Maybe `description`?

**Comment by luciecastella** [comment]
@danielhuppmann @phackstock

**Comment by danielhuppmann** [comment]
Yes, I would move to "description" as the default in new projects, this is also in line with the documentation [here](https://github.com/IAMconsortium/nomenclature/blob/main/doc/source/user_guide/variable.rst).

However, some (legacy) projects have neither of the two, so we should not make it a required attribute (yet).
